src/Microsoft.OData.Client/DataServiceContext.cs

```diff
@@ -253,6 +256,7 @@ internal DataServiceContext(Uri serviceRoot, ODataProtocolVersion maxProtocolVer
             this.httpStack = HttpStack.Auto;
             this.UsingDataServiceCollection = false;
             this.UsePostTunneling = false;
+            this.keyComparisonGeneratesFilterQuery = false;

```

Comment:
>false [](start = 53, length = 5)

Should this initially be true (for backward compatibility) and then set to false (or removed) in our next breaking change release (8.0)?